patches/chromium/skid-proof_video_decoding_memory_use.patch

```diff
@@ -0,0 +1,234 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Mack Straight <mack@discordapp.com>
+Date: Tue, 4 May 2021 03:23:02 -0700
+Subject: skid-proof video decoding memory use
+
+videos with inline resolution changes can use arbitrary amounts of
+memory to decode. upstream tried to fix this by checking for the OOM
+condition when allocating, but I found it to be inadequate -- the OOM
+crash just ends up happening elsewhere because we're still under severe
+memory pressure.
+
+instead, add a global memory budget for system memory allocated video
+frames. I set it to 768MB because that feels like it should be enough,
+but we can choose any value.
+
+re-evaluate current state of chromium once we upgrade and drop this fix
+if they put something better in.
+
+diff --git a/media/base/BUILD.gn b/media/base/BUILD.gn
+index 80cc2864762202bcf4211e8682d0fec3057bc655..3f2e359e92e74953a56f9b2b2880c07c12ff3646 100644
+--- a/media/base/BUILD.gn
++++ b/media/base/BUILD.gn
+@@ -313,6 +313,8 @@ jumbo_source_set("base") {
+     "video_frame.h",
+     "video_frame_layout.cc",
+     "video_frame_layout.h",
++    "video_frame_memory_budget.cc",
++    "video_frame_memory_budget.h",
+     "video_frame_metadata.cc",
+     "video_frame_metadata.h",
+     "video_frame_pool.cc",
+diff --git a/media/base/limits.h b/media/base/limits.h
+index 72980b35e65077162b9f4290592c36a00ce2f3b2..412d85d6536256bfc7f8d972426297d8c2517bb0 100644
+--- a/media/base/limits.h
++++ b/media/base/limits.h
+@@ -81,6 +81,11 @@ enum {
+       16,  // Matches ffmpeg's MAX_AUTO_THREADS. Higher values can result in
+            // immediate out of memory errors for high resolution content. See
+            // https://crbug.com/893984
++
++  // Maximum number of bytes to be used to allocate system memory buffers for
++  // video frames.
++  kVideoFrameBufferMemoryBudget = 1024 * 1024 * 768,
++  kVideoFrameBufferMaxAllocationSize = 4096 * 4096 * 3,
```

Comment:
this appears to be unchecked -- is it useful or should we nuke it?

if we keep it -- this *may* be a little conservative. (i'm thinking 8k video frames w/ 4 bytes per pixel -- not practical or common *right now* but 🤷🏻‍♂️.)

Reply:
checked @ https://github.com/discord/electron/pull/14/files#diff-597544b76f238329874271b455ae6f4c8a7511279cff485352ba61e31fa3b2e7R163 (the accessors are there for the tests I need to write :rofl:)

I put it at this rn because we appear to treat >= UHD videos as external links in the client, but yeah, we might want to adjust it in the future.